felles/src/test/kotlin/no/nav/familie/kontrakter/felles/DatoperiodeTest.kt

```diff
@@ -0,0 +1,243 @@
+        val periode = Datoperiode(LocalDate.of(2019, 1, 1), LocalDate.of(2019, 3, 1))
+
+        val overlapperIStartenAv = periodeSomErFør overlapperKunIStartenAv periode
+
+        overlapperIStartenAv shouldBe false
+    }
+
+    @Test
+    fun `overlapperIStartenAv returnerer false hvis denne perioden starter etter den som sendes inn`() {
+        val periodeSomErInneI = Datoperiode(LocalDate.of(2018, 9, 2), LocalDate.of(2018, 9, 25))
+        val periode = Datoperiode(LocalDate.of(2018, 9, 1), LocalDate.of(2018, 9, 1))
+
+        val overlapperIStartenAv = periodeSomErInneI overlapperKunIStartenAv periode
+
+        overlapperIStartenAv shouldBe false
+    }
+
+    @Test
+    fun `overlapperISluttenAv returnerer true hvis denne perioden overlapper i slutten av perioden som sendes inn`() {
+        val periodeSomOverlapperSlutten = Datoperiode(LocalDate.of(2019, 3, 1), LocalDate.of(2019, 3, 1))
+        val periode = Datoperiode(LocalDate.of(2019, 1, 1), LocalDate.of(2019, 3, 1))
+
+        val overlapperISluttenAv = periodeSomOverlapperSlutten overlapperKunISluttenAv periode
+
+        overlapperISluttenAv shouldBe true
+    }
+
+    @Test
+    fun `overlapperISluttenAv returnerer false hvis denne perioden er lik den som sendes inn`() {
+        val periodeSomErLik = Datoperiode(LocalDate.of(2019, 1, 1), LocalDate.of(2019, 3, 1))
+        val periode = Datoperiode(LocalDate.of(2019, 1, 1), LocalDate.of(2019, 3, 1))
+
+        val overlapperISluttenAv = periodeSomErLik overlapperKunISluttenAv periode
+
+        overlapperISluttenAv shouldBe false
+    }
+
+    @Test
+    fun `overlapperISluttenAv returnerer false hvis denne perioden er etter den som sendes inn`() {
+        val periodeSomErEtter = Datoperiode(LocalDate.of(2019, 4, 1), LocalDate.of(2019, 4, 1))
+        val periode = Datoperiode(LocalDate.of(2019, 1, 1), LocalDate.of(2019, 3, 1))
+
+        val overlapperISluttenAv = periodeSomErEtter overlapperKunISluttenAv periode
+
+        overlapperISluttenAv shouldBe false
+    }
+
+    @Test
+    fun `overlapperISluttenAv returnerer false hvis denne perioden slutter før den som sendes inn`() {
+        val periodeSomErInneI = Datoperiode(LocalDate.of(2018, 9, 2), LocalDate.of(2018, 9, 29))
+        val periode = Datoperiode(LocalDate.of(2018, 9, 1), LocalDate.of(2018, 9, 1))
+
+        val overlapperISluttenAv = periodeSomErInneI overlapperKunISluttenAv periode
+
+        overlapperISluttenAv shouldBe false
+    }
+
+    @Test
+    fun `lengdeIHeleMåneder feiler for perioder som ikke er hele måneder`() {
+        val periode = Datoperiode(LocalDate.of(2018, 9, 2), LocalDate.of(2020, 5, 31))
+
+        shouldThrowMessage(
+            "Forsøk på å beregne lengde i hele måneder for en periode som ikke er hele måneder: 2018-09-02 - 2020-05-31"
+        ) {
+            periode.lengdeIHeleMåneder()
+        }
+    }
+
+    @Test
+    fun `lengdeIHeleMåneder returnerer korrekt antall måneder`() {
```

Comment:
Kunne vi hatt en en enkel versjon -> happyday som returnerer 2, eller 3 så det er lett for meg å verifisere at antall er riktig?